model/map/src/main/java/org/keycloak/models/map/common/Serialization.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.common;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import java.io.IOException;
+
+/**
+ *
+ * @author hmlnarik
+ */
+public class Serialization {
+
+    public static final ObjectMapper MAPPER = new ObjectMapper();
```

Comment:
> Does it need to be public?

No, it need not, yet having it public helps in `ConcurrentHashMapStorageProvider` to use the same way of entity serialization as this "cloning" utility, and that in turn helps to debug issues with storing a representation of entities.

> Wondering if public would cause people misusing this instance and not using the method.

This instance should not be ordinarily visible to the public since model/map module (as well as any from the model/ modules) is not usually included as a dependency.

> I think I understand why you have another instance of `ObjectMapper` and `JsonSerialization` is not enough. But is there some way to just reuse `JsonSerialization` ?

Having these two separately is a better option, since the aim is to minimize impact on the existing code. The way of serialization is different (fields vs getters/setters), and aim is to have the entitites decoupled from serialization annotations. In the case of this `Serialization` class, jackson annotations on the fields (e.g. `@JsonIgnore`) are never in the entity, and are located in the serialization logics, i.e. in `Serialization` class. This makes it easy to decouple entities from their physical storage layout.